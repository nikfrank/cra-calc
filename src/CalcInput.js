import React, { Component } from 'react';

class CalcInput extends Component {
  render() {
    return (
      <div className="calc-input">
        {
          [1,2,3,4,5,6,7,8,9,0].map( i => (
            <button key={'number'+i}
                    className="calc-input-number"
                    onClick={()=>this.props.onNumber(i)}>
              {i}
            </button>
          ) )
        }
        {
          ['+', '-', '*', '/', '%'].map( f => (
            <button key={'func'+f}
                    className="calc-input-func"
                    onClick={()=>this.props.onFunc(f)}>
              {f}
            </button>
          ) )
        }
        <button onClick={this.props.onEq}
                className="calc-input-eq">
          =
        </button>
        <button onClick={this.props.onClear}
                className="calc-input-clear">
          C
        </button>
      </div>
    );
  }
}

export default CalcInput;
