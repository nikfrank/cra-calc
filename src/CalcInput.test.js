import React from 'react';
import ReactDOM from 'react-dom';
import CalcInput from './CalcInput';

import { shallow } from 'enzyme';

it('should render the input buttons', () => {
  const wrapper = shallow(
    <CalcInput/>
  );
  expect(wrapper).toMatchSnapshot();
});

it('calls the right functions', ()=>{
  // init spy functions
  const onNumber = jest.fn();
  const onFunc = jest.fn();
  const onClear = jest.fn();
  const onEq = jest.fn();

  // shallow render
  const wrapper = shallow(
    <CalcInput onNumber={onNumber}
               onFunc={onFunc}
               onClear={onClear}
               onEq={onEq}/>
  );

  // simulate clicks and make expectations
  
  wrapper.find('.calc-input-number')
         .first()
         .simulate('click');

  expect(onNumber).toBeCalledWith(1);

  
  wrapper.find('.calc-input-func')
         .at(4)
         .simulate('click');

  expect(onFunc).toBeCalledWith('%');


  wrapper.find('.calc-input-eq')
         .simulate('click');

  expect(onEq.mock.calls.length).toEqual(1);


  const clearBtn = wrapper.find('.calc-input-clear');

  clearBtn.simulate('click');
  clearBtn.simulate('click');

  expect(onClear.mock.calls.length).toEqual(2);
});
