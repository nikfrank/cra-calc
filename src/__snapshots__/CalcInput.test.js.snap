exports[`test should render the input buttons 1`] = `
<div
  className="calc-input">
  <button
    className="calc-input-number"
    onClick={[Function]}>
    1
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    2
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    3
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    4
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    5
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    6
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    7
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    8
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    9
  </button>
  <button
    className="calc-input-number"
    onClick={[Function]}>
    0
  </button>
  <button
    className="calc-input-func"
    onClick={[Function]}>
    +
  </button>
  <button
    className="calc-input-func"
    onClick={[Function]}>
    -
  </button>
  <button
    className="calc-input-func"
    onClick={[Function]}>
    *
  </button>
  <button
    className="calc-input-func"
    onClick={[Function]}>
    /
  </button>
  <button
    className="calc-input-func"
    onClick={[Function]}>
    %
  </button>
  <button
    className="calc-input-eq">
    =
  </button>
  <button
    className="calc-input-clear">
    C
  </button>
</div>
`;
